SourceCode/ChaosRayTracing/ChaosRayTracing/ThreadPool.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <mutex>
+#include <queue>
+#include <functional>
+#include <vector>
+#include <condition_variable>
+#include <future>
+
+class ThreadPool
```

Comment:
Nice idea for the thread pool

Reply:
Thanks, I reused the thread pool that I implemented for the homework in the algorithms session.